File: src/main/java/jwang/y2022/Day1.java
package jwang.y2022;

import utils.Utils;

import java.util.List;

public class Day1 {

  public static void main(String... args) throws Exception {
    List<String> data = Utils.readFromFile("src/main/resources/jwang/y2022/Day1Input.txt");
    Maxes maxes = new Maxes();
    int max = 0;
    int sum = 0;
    for (String line : data) {
      if (line == null || line.length() == 0) {
        maxes.compareAndUpdate(sum);
        if (sum > max) {
          max = sum;
        }
        sum = 0;
      }
      else {
        sum += Integer.parseInt(line);
      }
    }
    maxes.compareAndUpdate(sum);
    System.out.println(max);
    System.out.println(maxes.total());
  }

  static class Maxes {
    int first = 0;
    int second = 0;
    int third = 0;

    void compareAndUpdate(int val) {
      if (val > first) {
        third = second;
        second = first;
        first = val;
      } else if (val > second) {
        third = second;
        second = val;
      } else if (val > third) {
        third = val;
      }
    }

    int total() {
      return first + second + third;
    }
  }
}

File: src/main/rust/jwang/y2022/day1.rs

use std::fs::File;
use std::io::{BufReader, BufRead};

pub fn day1() {
    let test: Vec<&str> = vec![
        "1000",
        "2000",
        "3000",
        "",
        "4000",
        "",
        "5000",
        "6000",
        "",
        "7000",
        "8000",
        "9000",
        "",
        "10000"
    ];

    let filename = ".\\src\\main\\resources\\jwang\\y2022\\Day1Input.txt";
    let file = File::open(filename).unwrap();
    let reader = BufReader::new(file);

    let mut max = 0;
    let mut sum = 0;
    for result in reader.lines() {
        let line : String = result.unwrap();
    // for line in test {
        println!("{:?}", line);
        if line.is_empty() {
            if sum > max {
                max = sum;
                sum = 0;
            }
        }
        else {
            sum += line.parse::<i32>().unwrap();
        }
    }
    println!("{:?}", max);
}